share/functions/__fish_svn_prompt.fish

```diff
@@ -0,0 +1,169 @@
+# colour of the revision number to display in the prompt
+set -g __fish_svn_prompt_color_revision yellow
+
+# setting the prompt status separator character
+set -g __fish_svn_prompt_char_separator "|"
+
+# ==============================
+
+# SVN status symbols
+# Do not change these! These are the expected characters that are output from `svn status`, they are taken from `svn help status`
```

Comment:
I don't think setting them as local variables is going to take that much time. In my experience most fishscript is taking the most time in calls to external tools (the hg_prompt was particularly bad about that, to the point that we've added custom code to find the .hg directory). Even though svn seems reasonably quick, it's still a few forks.

My gut tells me that optimization here is achieved by replacing `grep` or `math` (which forks to `bc`), not via defining variables.